gradle-baseline-java/src/main/java/com/palantir/baseline/tasks/CheckUnusedDependenciesTask.java

```diff
@@ -16,76 +16,86 @@
 
 package com.palantir.baseline.tasks;
 
-import com.google.common.base.Preconditions;
 import com.google.common.collect.Streams;
 import com.palantir.baseline.plugins.BaselineExactDependencies;
 import com.palantir.gradle.failurereports.exceptions.ExceptionWithSuggestion;
+import java.io.Serializable;
 import java.nio.file.Path;
-import java.util.Collections;
 import java.util.Comparator;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.inject.Inject;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.DependencyArtifact;
+import org.gradle.api.artifacts.ModuleDependency;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
 import org.gradle.api.artifacts.ResolvedArtifact;
 import org.gradle.api.artifacts.ResolvedConfiguration;
-import org.gradle.api.file.FileCollection;
+import org.gradle.api.attributes.Usage;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.model.ObjectFactory;
 import org.gradle.api.provider.ListProperty;
-import org.gradle.api.provider.Property;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.provider.SetProperty;
 import org.gradle.api.tasks.Classpath;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.annotations.NotNull;
+import org.jspecify.annotations.Nullable;
 
-@SuppressWarnings("for-rollout:NonAbstractGradleType")
-public class CheckUnusedDependenciesTask extends DefaultTask {
+public abstract class CheckUnusedDependenciesTask extends DefaultTask {
 
-    @SuppressWarnings("for-rollout:GradleTypesAsFields")
-    private final ListProperty<Configuration> dependenciesConfigurations;
+    @Input
+    public abstract SetProperty<String> getIgnored();
+
+    @Classpath
+    public abstract ListProperty<Configuration> getDependenciesConfigurations();
```

Comment:
Just to note, this is still non-configuration cacheable/won't run in parallel as it takes `Configuration` - but this was there before, best to fix in a later PR.